libs/components/src/lib/video-player/video-player.spec.ts

```diff
@@ -0,0 +1,286 @@
+			await elementUpdated(element);
+			const bigPlayBtn = getBigPlayButton();
+			expect(bigPlayBtn?.classList.contains('vjs-hidden')).toBe(false);
+		});
+
+		it('should allow the src to be updated', async () => {
+			element = (await fixture(
+				`<${COMPONENT_TAG} autoplay>
+					<source src="${VIDEO_SRC}" type="video/mp4">
+				</${COMPONENT_TAG}>`
+			)) as VideoPlayer;
+			await elementUpdated(element);
+			element.src = 'new-src.mp4';
+			await elementUpdated(element);
+			const bigPlayBtn = getBigPlayButton();
+			expect(bigPlayBtn?.classList.contains('vjs-hidden')).toBe(false);
+		});
+	});
+
+	describe('autoplay', () => {
+		it('should set the muted attribute on the video element', async () => {
+			element = (await fixture(
+				`<${COMPONENT_TAG} autoplay>
+					<source src="${VIDEO_SRC}" type="video/mp4">
+				</${COMPONENT_TAG}>`
+			)) as VideoPlayer;
+			const videoEle = getVideoEle();
+			expect(videoEle?.autoplay).toBe(true);
+		});
+	});
+
+	describe('loop', () => {
+		it('should set the loop attribute on the video element', async () => {
+			element = (await fixture(
+				`<${COMPONENT_TAG} loop>
+					<source src="${VIDEO_SRC}" type="video/mp4">
+				</${COMPONENT_TAG}>`
+			)) as VideoPlayer;
+			const videoEle = getVideoEle();
+			expect(videoEle?.loop).toBe(true);
+		});
+	});
+
+	describe('playback rates', () => {
+		it('should set custom playback rates', async () => {
+			element = (await fixture(
+				`<${COMPONENT_TAG} playback-rates="0.25, 0.5, 1">
+					<source src="${VIDEO_SRC}" type="video/mp4">
+				</${COMPONENT_TAG}>`
+			)) as VideoPlayer;
+			await elementUpdated(element);
+			const playbackRatesMenuItems = element.shadowRoot?.querySelectorAll('.vjs-playback-rate .vjs-menu li');
+			expect(playbackRatesMenuItems?.length).toBe(3);
+			if (playbackRatesMenuItems?.length === 3) {
+				expect(playbackRatesMenuItems[0].querySelector('.vjs-menu-item-text')!.textContent).toBe('1x');
+				expect(playbackRatesMenuItems[1].querySelector('.vjs-menu-item-text')!.textContent).toBe('0.5x');
+				expect(playbackRatesMenuItems[2].querySelector('.vjs-menu-item-text')!.textContent).toBe('0.25x');
+			}
+		});
+
+		it('should disable playback rates when passed an empty string', async () => {
+			element = (await fixture(
+				`<${COMPONENT_TAG} playback-rates="">
+					<source src="${VIDEO_SRC}" type="video/mp4">
+				</${COMPONENT_TAG}>`
+			)) as VideoPlayer;
+			const playbackRate = element.shadowRoot?.querySelector('.vjs-playback-rate');
+			expect(playbackRate?.classList.contains('vjs-hidden')).toBe(true);
+		});
+	});
+
+	describe('captions', () => {
+		it('displays the captions button and menu', async () => {
+			element = (await fixture(
+				`<${COMPONENT_TAG} playback-rates="">
+					<source src="${VIDEO_SRC}" type="video/mp4">
+					<track 
+						kind="captions"
+						src="https://d2zihajmogu5jn.cloudfront.net/elephantsdream/captions.en.vtt" 
+						srclang="en" 
+						label="English" 
+						default>
```

Comment:
I suggest not to use external content. This can actually be replaced with any other string (even an empty one) because what matters is the number of tracks of `kind="captions"`.
```suggestion
						src=""
```